src/cephadm/cephadm

```diff
@@ -985,6 +985,34 @@ def check_ip_port(ctx, ip, port):
         except OSError as e:
             raise Error(e)
 
+def _parse_ss_lpnt(data:str):
+    # ss doen't support json output. Thus we have to parse it.
+    # https://unix.stackexchange.com/questions/157823/list-ports-a-process-pid-is-listening-on-preferably-using-iproute2-tools
+    out = []
+    for line in data.splitlines():
+        if line.startswith('STATE'):
+            if ' '.join(line.split()) != 'State Recv-Q Send-Q Local Address:Port Peer Address:Port':
+                logger.warning('ss -lpnt: unable to parse header "{line}"')
+                return {}
+        if 'users:((' not in line:
+            continue
+        before, after = line.split('users:((')
+        data = dict(zip('State,Recv-Q,Send-Q,Local Address:Port,Peer Address:Port'.split(','), before.split()))
+
+        users = after.rstrip('))').split(',')
+        data.update(dict(zip('name pid'.split(), users[:2])))
+        out.append(data)
+    return out
+
+def port_in_use_by(ctx, port_num: int) -> str:
```

Comment:
I suggest to use re.
ex:
```python
def port_used_by( port_num: int):
    out, err, code = call(ctx, ['ss', '-l', '-p', '-n', '-t'])
    pattern = '\d+:' + port_num + '\s+\d+.\d+.\d+.\d+.+users:\(\(\"(?P<process>.*)\",'
    used = re.search(pattern, out)
    if used:
        return used.group('process')
     else:
         return  ""
```

Reply:
building something that also works for IPv6 is going to be painful. and in the end I'd opt for a reg ex that simply splits by `\s`